flow/aggregates.py

```diff
@@ -206,30 +206,24 @@ def aggregator_function(jobs):
             for key, group in groupby(sorted(jobs, key=keyfunction), key=keyfunction):
                 yield group
 
-        return cls(aggregator_function, sort_by, reverse_order, select)
+        return cls(aggregator_function, sort_by, sort_ascending, select)
 
     def __eq__(self, other):
-        return type(self) == type(other) and \
-               not self._is_aggregate and not other._is_aggregate
+        return type(self) == type(other) and not self._is_aggregate and not other._is_aggregate
 
     def __hash__(self):
-        blob_l = [
-            hash(self._sort_by), hash(self._reverse_order), hash(self._is_aggregate),
-            self._get_unique_function_id(self._aggregator_function),
-            self._get_unique_function_id(self._select)
-        ]
-        blob = ','.join(str(attr) for attr in blob_l)
-
-        return int(_hash(blob), 16)
+        return hash((self._sort_by, self._sort_ascending, self._is_aggregate,
+                     self._get_unique_function_id(self._aggregator_function),
+                     self._get_unique_function_id(self._select)))
 
     def _get_unique_function_id(self, func):
         """Generate unique id for the function passed. The id returned is used to generate
         hash and compare arbitrary types which are callable like ``self._aggregator_function``
         and ``self._select`` attributes.
         """
         try:
-            return func.__code__.co_code
-        except Exception:  # Got something other than a function
+            return hash(func.__code__.co_code)
+        except AttributeError:  # Cannot access function's compiled bytecode
```

Comment:
We should hash the bytecode before returning, since this is purely used as a unique identifier. We also shouldn't catch `Exception`! We know this should raise an `AttributeError` if it can't access the attribute.

Reply:
See my comment above (where this function is used).